File: AppGestion/api/src/main/java/com/appgestion/api/catalog/RubroAutonomoCatalog.java
package com.appgestion.api.catalog;

import java.util.Set;

/**
 * Códigos permitidos para {@code Empresa.rubroAutonomoCodigo} (métricas; no en facturas).
 * Debe coincidir con {@code rubro-autonomo.catalog.ts} en el frontend.
 */
public final class RubroAutonomoCatalog {

    private static final Set<String> CODIGOS = Set.of(
            // Informática y digital
            "DESARROLLO_SOFTWARE",
            "DISENO_WEB_UX",
            "CONSULTORIA_IT",
            "CIBERSEGURIDAD",
            "MARKETING_DIGITAL",
            "COMMUNITY_MANAGER",
            // Construcción y reformas
            "ALBANILERIA",
            "FONTANERIA",
            "ELECTRICIDAD",
            "CARPINTERIA",
            "PINTURA",
            "REFORMAS_INTEGRALES",
            "INSTALACIONES_CLIMATIZACION",
            // Comercio
            "COMERCIO_MINORISTA",
            "COMERCIO_ONLINE",
            "DISTRIBUCION",
            // Hostelería
            "RESTAURACION",
            "BAR_CAFETERIA",
            "CATERING",
            "PASTELERIA",
            // Salud y bienestar
            "FISIOTERAPIA",
            "ENFERMERIA_LIBERAL",
            "PSICOLOGIA",
            "NUTRICION_DIETETICA",
            "PODOLOGIA",
            "FARMACIA",
            // Educación y formación
            "FORMACION_PROFESIONAL",
            "IDIOMAS",
            "REFUERZO_ESCOLAR",
            "COACHING",
            // Transporte y motor
            "TAXI_VTC",
            "TRANSPORTE_MERCANCIAS",
            "MECANICA_VEHICULOS",
            "MOTO_TALLER",
            // Servicios profesionales y admin
            "ASESORIA_FISCAL_LABORAL",
            "ABOGACIA",
            "GESTORIA_ADMINISTRATIVA",
            "ARQUITECTURA",
            "INGENIERIA_TECNICA",
            "MEDIACION",
            // Arte, foto y audiovisual
            "FOTOGRAFIA",
            "VIDEO_AUDIOVISUAL",
            "DISENO_GRAFICO",
            "ILUSTRACION",
            // Belleza y cuidado personal
            "PELUQUERIA",
            "ESTETICA",
            "UNAS_MICROPIGMENTACION",
            // Limpieza, jardín y mantenimiento
            "LIMPIEZA",
            "JARDINERIA",
            "MANTENIMIENTO_EDIFICIOS",
            "CONTROL_DE_PLAGAS",
            // Agricultura y ganadería
            "AGRICULTURA_GANADERIA",
            "VITICULTURA_ENOLOGIA",
            // Otros
            "OTRO",
            "PREFIERO_NO_DECIR"
    );

    private RubroAutonomoCatalog() {}

    public static boolean esCodigoValido(String codigo) {
        return codigo != null && CODIGOS.contains(codigo);
    }
}

File: AppGestion/frontend/src/app/features/cuenta/datos-empresa/rubro-autonomo.catalog.ts
/**
 * Rubros de autónomos por categoría (códigos estables para métricas en servidor).
 * Debe coincidir con {@link com.appgestion.api.catalog.RubroAutonomoCatalog} en la API.
 */
export interface RubroAutonomoOpcion {
  codigo: string;
  etiqueta: string;
}

export interface RubroAutonomoCategoria {
  id: string;
  etiqueta: string;
  opciones: RubroAutonomoOpcion[];
}

export const RUBRO_AUTONOMO_CATEGORIAS: RubroAutonomoCategoria[] = [
  {
    id: 'informatica',
    etiqueta: 'Informática y digital',
    opciones: [
      { codigo: 'DESARROLLO_SOFTWARE', etiqueta: 'Desarrollo de software' },
      { codigo: 'DISENO_WEB_UX', etiqueta: 'Diseño web / UX' },
      { codigo: 'CONSULTORIA_IT', etiqueta: 'Consultoría informática' },
      { codigo: 'CIBERSEGURIDAD', etiqueta: 'Ciberseguridad' },
      { codigo: 'MARKETING_DIGITAL', etiqueta: 'Marketing digital' },
      { codigo: 'COMMUNITY_MANAGER', etiqueta: 'Community manager' },
    ],
  },
  {
    id: 'construccion',
    etiqueta: 'Construcción y reformas',
    opciones: [
      { codigo: 'ALBANILERIA', etiqueta: 'Albañilería' },
      { codigo: 'FONTANERIA', etiqueta: 'Fontanería' },
      { codigo: 'ELECTRICIDAD', etiqueta: 'Electricidad' },
      { codigo: 'CARPINTERIA', etiqueta: 'Carpintería' },
      { codigo: 'PINTURA', etiqueta: 'Pintura' },
      { codigo: 'REFORMAS_INTEGRALES', etiqueta: 'Reformas integrales' },
      { codigo: 'INSTALACIONES_CLIMATIZACION', etiqueta: 'Climatización / frío industrial' },
    ],
  },
  {
    id: 'comercio',
    etiqueta: 'Comercio',
    opciones: [
      { codigo: 'COMERCIO_MINORISTA', etiqueta: 'Comercio minorista (tienda física)' },
      { codigo: 'COMERCIO_ONLINE', etiqueta: 'Comercio online / e‑commerce' },
      { codigo: 'DISTRIBUCION', etiqueta: 'Distribución / mayorista' },
    ],
  },
  {
    id: 'hosteleria',
    etiqueta: 'Hostelería',
    opciones: [
      { codigo: 'RESTAURACION', etiqueta: 'Restauración' },
      { codigo: 'BAR_CAFETERIA', etiqueta: 'Bar / cafetería' },
      { codigo: 'CATERING', etiqueta: 'Catering' },
      { codigo: 'PASTELERIA', etiqueta: 'Pastelería / panadería artesanal' },
    ],
  },
  {
    id: 'salud',
    etiqueta: 'Salud y bienestar',
    opciones: [
      { codigo: 'FISIOTERAPIA', etiqueta: 'Fisioterapia' },
      { codigo: 'ENFERMERIA_LIBERAL', etiqueta: 'Enfermería (ejercicio liberal)' },
      { codigo: 'PSICOLOGIA', etiqueta: 'Psicología' },
      { codigo: 'NUTRICION_DIETETICA', etiqueta: 'Nutrición / dietética' },
      { codigo: 'PODOLOGIA', etiqueta: 'Podología' },
      { codigo: 'FARMACIA', etiqueta: 'Farmacia (titular)' },
    ],
  },
  {
    id: 'educacion',
    etiqueta: 'Educación y formación',
    opciones: [
      { codigo: 'FORMACION_PROFESIONAL', etiqueta: 'Formación profesional / cursos' },
      { codigo: 'IDIOMAS', etiqueta: 'Idiomas' },
      { codigo: 'REFUERZO_ESCOLAR', etiqueta: 'Refuerzo escolar' },
      { codigo: 'COACHING', etiqueta: 'Coaching / desarrollo personal' },
    ],
  },
  {
    id: 'transporte',
    etiqueta: 'Transporte y motor',
    opciones: [
      { codigo: 'TAXI_VTC', etiqueta: 'Taxi / VTC' },
      { codigo: 'TRANSPORTE_MERCANCIAS', etiqueta: 'Transporte de mercancías' },
      { codigo: 'MECANICA_VEHICULOS', etiqueta: 'Mecánica de vehículos' },
      { codigo: 'MOTO_TALLER', etiqueta: 'Taller de motos' },
    ],
  },
  {
    id: 'profesional',
    etiqueta: 'Servicios profesionales y administración',
    opciones: [
      { codigo: 'ASESORIA_FISCAL_LABORAL', etiqueta: 'Asesoría fiscal / laboral' },
      { codigo: 'ABOGACIA', etiqueta: 'Abogacía' },
      { codigo: 'GESTORIA_ADMINISTRATIVA', etiqueta: 'Gestoría administrativa' },
      { codigo: 'ARQUITECTURA', etiqueta: 'Arquitectura' },
      { codigo: 'INGENIERIA_TECNICA', etiqueta: 'Ingeniería técnica' },
      { codigo: 'MEDIACION', etiqueta: 'Mediación / arbitraje' },
    ],
  },
  {
    id: 'arte',
    etiqueta: 'Arte, foto y audiovisual',
    opciones: [
      { codigo: 'FOTOGRAFIA', etiqueta: 'Fotografía' },
      { codigo: 'VIDEO_AUDIOVISUAL', etiqueta: 'Vídeo / audiovisual' },
      { codigo: 'DISENO_GRAFICO', etiqueta: 'Diseño gráfico' },
      { codigo: 'ILUSTRACION', etiqueta: 'Ilustración' },
    ],
  },
  {
    id: 'belleza',
    etiqueta: 'Belleza y estética',
    opciones: [
      { codigo: 'PELUQUERIA', etiqueta: 'Peluquería' },
      { codigo: 'ESTETICA', etiqueta: 'Estética / tratamientos' },
      { codigo: 'UNAS_MICROPIGMENTACION', etiqueta: 'Uñas / micropigmentación' },
    ],
  },
  {
    id: 'limpieza',
    etiqueta: 'Limpieza y mantenimiento',
    opciones: [
      { codigo: 'LIMPIEZA', etiqueta: 'Limpieza (locales, oficinas…)' },
      { codigo: 'JARDINERIA', etiqueta: 'Jardinería' },
      { codigo: 'MANTENIMIENTO_EDIFICIOS', etiqueta: 'Mantenimiento de edificios' },
      { codigo: 'CONTROL_DE_PLAGAS', etiqueta: 'Control de plagas' },
    ],
  },
  {
    id: 'agricultura',
    etiqueta: 'Agricultura y sector primario',
    opciones: [
      { codigo: 'AGRICULTURA_GANADERIA', etiqueta: 'Agricultura / ganadería' },
      { codigo: 'VITICULTURA_ENOLOGIA', etiqueta: 'Viticultura / enología' },
    ],
  },
  {
    id: 'otros',
    etiqueta: 'Otros',
    opciones: [
      { codigo: 'OTRO', etiqueta: 'Otro (genérico)' },
      { codigo: 'PREFIERO_NO_DECIR', etiqueta: 'Prefiero no indicarlo' },
    ],
  },
];
